src/main/resources/swagger/ga4gh-tool-discovery.yaml

```diff
@@ -614,6 +614,87 @@ definitions:
         description: >-
           Source of metadata that can support a verified tool, such as an email
           or URL
+      inputs:
+        description: A list of inputs for this tool
+        type: array
+        items:
+          $ref: '#/definitions/ToolParameter'
+  ToolParameter:
+    type: object
+    description: >-
+      An input parameter for a tool or workflow.
+    required:
+      - name
+      - valueType
+      - optional
+    properties:
+      name:
+        type: string
+        description: The name of this input value (formatted as expected by the tool)
+      valueType:
+        $ref: '#/definitions/ValueType'
+      optional:
+        type: boolean
+        description: Whether the tool allows this value to not be specified
+      default:
+        type: string
+        description: The in-language expression used to evaluate a default value for this parameter, if
+          none is supplied.
+  ValueType:
+    description: The type expected for a given value.
+    type: object
+    required:
+      - valueType
+    maxProperties: 2
+    properties:
+      valueType:
+        type: string
+        description: The type of this value
+        enum:
+          - String
+          - File
+          - Directory
+          - Float
+          - Int
+          - Boolean
+          - Optional
+          - Array
+          - Tuple
+          - Map
+          - Object
+      optionalType:
```

Comment:
Unfortunately, I don't think there's a way to say `optionalType` is required if and only if the valueType is `Optional`

Reply:
Is an interesting thought, as long as the swagger 2.0 version is generally compatible with additional optional extensions written in OpenApi 3.0, this approach could work. 

We just have to figure out a way to avoid having the automatic swagger 2.0 to OpenApi 3.0 avoid overwriting the 3.0 additions while transferring over legitimate changes.

Reply:
@cjllanwarne re-reading, it doesn't seem like this would be a blocker. 
(i.e. describe the desired behavior in comments until we migrate to OpenApi 3.0)